source/grid/manifold.cc

```diff
@@ -144,22 +127,24 @@ get_new_point (const std::vector<Point<spacedim> > &surrounding_points,
 template <int dim, int spacedim>
 void
 Manifold<dim, spacedim>::
-add_new_points (const std::vector<Point<spacedim> > &surrounding_points,
-                const Table<2,double>               &weights,
-                std::vector<Point<spacedim> >       &new_points) const
+add_new_points (const ArrayView<const Point<spacedim>> &surrounding_points,
+                const Table<2,double>                  &weights,
+                ArrayView<Point<spacedim>>              new_points) const
 {
   AssertDimension(surrounding_points.size(), weights.size(1));
-  Assert(&surrounding_points != &new_points,
-         ExcMessage("surrounding_points and new_points cannot be the same "
-                    "array"));
 
-  const unsigned int n_points = surrounding_points.size();
-  std::vector<double> local_weights(n_points);
+  const std::size_t n_points = surrounding_points.size();
+  // TODO find a better dimension-dependent estimate for the size of this
+  // vector
+  boost::container::small_vector<double, 20> local_weights(n_points);
```

Comment:
I'm not sure we know that. It really depends on the degree of interpolation (or whatever comes in here). In the end, the allocation is a one-time cost that must be paid off by enough operations. I would have chosen a much larger value here, maybe 200 or 500.